plugin/mpi/PETSc-code.hpp

```diff
@@ -4705,10 +4730,19 @@ namespace PETSc {
           if (op) {
             ffassert(op);
             const OneOperator* codeM =
-              op->Find("(", ArrayOfaType(atype< long >( ), atype< double >( ), atype< double >( ),
-                                         atype< double >( ), atype< KN< PetscScalar >* >( ),
-                                         atype< KN< PetscScalar >* >( ), false));
-            user->conv = new NonlinearSolver< Type >::IConvF_O(in->n, stack, codeM);
+              op->Find("(", ArrayOfaType(atype< long >( ), atype< double >( ),
+                                          atype< KN< PetscScalar >* >( ), false));
+            user->mon = new NonlinearSolver< Type >::IMonF_O(in->n, stack, codeM);
+            SNESMonitorSet(snes, Monitor< NonlinearSolver< Type > >, &user, NULL);
+          }
+          const Polymorphic* op2 = nargs[12] ? dynamic_cast< const Polymorphic* >(nargs[12]) : nullptr;
+          if (op2) {
+            ffassert(op2);
+            const OneOperator* codeC =
+              op2->Find("(", ArrayOfaType(atype< long >( ), atype< double >( ), atype< double >( ),
```

Comment:
```suggestion
          op = nargs[12] ? dynamic_cast< const Polymorphic* >(nargs[12]) : nullptr;
          if (op) {
            ffassert(op);
            const OneOperator* codeC =
              op->Find("(", ArrayOfaType(atype< long >( ), atype< double >( ), atype< double >( ),
```
I'm not sure there is a real need for a new `Polymorphic*` variable here, may as well reuse `op`?